filters/openpolicyagent/evaluation.go

```
@@ -16,17 +16,20 @@ import (
 )
 
 func (opa *OpenPolicyAgentInstance) Eval(ctx context.Context, req *ext_authz_v3.CheckRequest) (*envoyauth.EvalResult, error) {
-	result, stopeval, err := envoyauth.NewEvalResult()
-	span := opentracing.SpanFromContext(ctx)
-	if span != nil {
-		span.SetTag("opa.decision_id", result.DecisionID)
-	}
+
+	decisionId, err := opa.idGenerator.Generate()
 
 	if err != nil {
 		opa.Logger().WithFields(map[string]interface{}{"err": err}).Error("Unable to generate decision ID.")
 		return nil, err
 	}
 
+	result, stopeval, err := envoyauth.NewEvalResult(WithDecisionID(decisionId))
```

This error is not checked now